lib/itamae/list.rb

```diff
@@ -0,0 +1,69 @@
+module Itamae
+  class List
+    def run
+      require 'rubygems'
+      require 'rubygems/exceptions'
+      require 'pathname'
+
+      pattern = /^itamae-plugin-recipe/
+      # from rubygems Gem::Commands::QueryCommand#show_local_gems
+      specs = Gem::Specification.find_all do |s|
+        s.name =~ pattern
+      end
+
+      req = Gem::Requirement.default
+      dep = Gem::Deprecate.skip_during { Gem::Dependency.new pattern, req }
+      specs.select! do |s|
+        dep.match?(s.name, s.version, false)
+      end
+      spec_tuples = specs.map do |spec|
+        [spec.name_tuple, spec]
+      end
+
+      # from rubygems Gem::Commands::QueryCommand#output_query_results
+      versions = Hash.new { |h,name| h[name] = [] }
+      spec_tuples.each do |spec_tuple, source|
+        versions[spec_tuple.name] << [spec_tuple, source]
+      end
+      versions = versions.sort_by do |(n,_),_|
+        n.downcase
+      end
+
+      # from rubygems Gem::Commands::QueryCommand#output_versions
+      versions.each do |gem_name, matching_tuples|
```

Comment:
Is `gem_name` unused? I think you should add a prefix `_` if that.

Reply:
Thank you for your pointing out! I did so at 80fadc17358a75503497a943f5412287f89e403d